x-pack/plugins/canvas/server/saved_objects/workpad_references.ts

```diff
@@ -6,14 +6,15 @@
  */
 
 import { fromExpression, toExpression } from '@kbn/interpreter/common';
+import { PersistableStateService } from '../../../../../src/plugins/kibana_utils/common';
 import { SavedObjectReference } from '../../../../../src/core/server';
 import { WorkpadAttributes } from '../routes/workpad/workpad_attributes';
 
-import { ExpressionsServerSetup } from '../../../../../src/plugins/expressions/server';
+import type { ExpressionAstExpression } from '../../../../../src/plugins/expressions';
 
 export const extractReferences = (
   workpad: WorkpadAttributes,
-  expressions: ExpressionsServerSetup
+  expressions: PersistableStateService<ExpressionAstExpression>
```

Comment:
why is this needed ? looks the previous state was correct

Reply:
No, that was not correct because `inject` and `extract` are parts of the `PersistableStateService` interface. [Interface segregation](https://en.wikipedia.org/wiki/Interface_segregation_principle) is a good practice because then I could've refactored the `ExpressionsService` without touching this code.

Reply:
makes me feel like canvas needs to know too much about expression service now (the fact that it implements `PersistableStateService<ExpressionAstExpression>`). 

cc @stacey-gammon what do you think ?

Reply:
Change looks okay to me. I think this function already needs to know that the expression service implements PersistableStateService, because that's the only reason it's calling `extract` on it.